### PR TITLE
tracee-ebpf: add MemProtAlert type to external package

### PR DESCRIPTION
**Pull request by yanivagman**
Not too long ago we had "security-alerts" as part of tracee-ebpf (until #1114 removed it).
"alerts" where messages sent by an eBPF event as an extra argument, to pass some information about a security alert.
Introduce MemProtAlert in the external package, so it can be used by tracee-ebpf consumers

This will also allow us to expose "alerts" as raw arguments (related #1123 )

### CODE REVIEW

File: tracee-ebpf/tracee/argprinters.go [hunk 1]
```diff
@@ -53,17 +55,15 @@ func (t *Tracee) prepareArgs(ctx *context, args map[string]interface{}) error {
...
-		}
-	}
-
 	// EventIDs which are optionally parsed/enriched into human readable formats
 	if !t.config.Output.ParseArguments {
 		return nil
 	}
 	switch ctx.EventID {
+	case MemProtAlertEventID:
+		if message, isMessage := args["alert"].(external.Message); isMessage {
```
Comment (itaysk) [inline tracee-ebpf/tracee/argprinters.go:64]: message?

**Review by itaysk — COMMENTED** [review]
Good that you raise this an fix it. I think it should be called event not message. We use event in our terminology to represent something tracee-ebpf emits.

**Comment by itaysk** [comment]
related: https://github.com/aquasecurity/tracee/discussions/1105

**Comment by yanivagman** [comment]
This is not an event. The "message" concept presented here is an argument inside an event.
The current use case is of the "mem_prot_alert" event, where the only argument is the alert.
But this concept can be used by any event, to decode different messages as an argument of the event

File: tracee-ebpf/tracee/argprinters.go (hunk 1 above)
Comment (yanivagman) [reply tracee-ebpf/tracee/argprinters.go:64]: No. "alert" is the name of the argument, and I think we should keep this name

**Comment by itaysk** [comment]
ok so I misunderstood your intention. it's just about encoding the strings passed as event arguments?
`mem_prot_alert` is not an event not message, but it's `alert` argument is a message?

**Comment by yanivagman** [comment]
```
ok so I misunderstood your intention. it's just about encoding the strings passed as event arguments?
mem_prot_alert is not an event not message, but it's alert argument is a message?
```

Yes, it's about encoding the strings as event arguments. It then can be used by external consumer as a raw argument (without "stringifying" it), or given as a parsed string.

mem_prot_alert is a special event, which has only one argument called "alert" which is of message type. In the future, we can think of other abstractions, such as what you suggested in #1105

**Comment by itaysk** [comment]
does all messages need `Ts` and `Payload`? It seems to me that instead of generalizing `security_alert`, what we needed was the opposite, to make it more specific no? (e.g `mem_prot_alert_message` or something like this)
Take a look at https://github.com/aquasecurity/libbpfgo/pull/89#discussion_r750007493 and tell me what you think

**Comment by yanivagman** [comment]
> does all messages need Ts and Payload? It seems to me that instead of generalizing security_alert, what we needed was the opposite, to make it more specific no? (e.g mem_prot_alert_message or something like this) like commit creds
> Take a look at aquasecurity/libbpfgo#89 (comment) and tell me what you think

I took the extra step and removed Ts and Payload.
"message" is now an enum exposed by external, which can be translated to a string when parse-arguments is chosen

**Review by itaysk — APPROVED** [review]

**Pull request merged**